### PR TITLE
[wingman -> rllib] Improved stats changes in AsyncSamplesOptimizer

### PR DESCRIPTION
**Pull request by bjg2**
Stats gathering in AsyncSamplesOptimizer is improved, as samples_throughput and train_throughput were giving bad values (mostly 0, with periodical peaks, instead of a steady graph showing throughput per second).

### CODE REVIEW

File: python/ray/rllib/optimizers/async_samples_optimizer.py [hunk 1]
```diff
@@ -85,7 +84,9 @@ def _init(self,
         self.timers = {k: TimerStat() for k in ["train", "sample"]}
```
Comment (ericl) [inline python/ray/rllib/optimizers/async_samples_optimizer.py]: Does it also make sense to remove the timers?

Comment (bjg2) [reply python/ray/rllib/optimizers/async_samples_optimizer.py]: Well, I think so. Replaced self.timers with self._optimizer_step_timer.

**Review by ericl — APPROVED** [review]
Just tried out it, looks good.

**Comment by ericl** [comment]
Pushed some lint fixes. Note that num remote_evaluators should be >0 still, since async samples optimizer doesn't have fallback code for 0 workers case like other optimizers do.

File: python/ray/rllib/optimizers/async_samples_optimizer.py [hunk 2]
```diff
@@ -130,27 +152,24 @@ def stop(self):
...
-            "sample_throughput": round(self.timers["sample"].mean_throughput,
-                                       3),
-            "train_throughput": round(self.timers["train"].mean_throughput, 3),
+        stats = dict({
             "num_weight_syncs": self.num_weight_syncs,
             "num_steps_replayed": self.num_replayed,
             "timing_breakdown": timing,
             "learner_queue": self.learner.learner_queue_size.stats(),
-        }
+        }, **self.get_mean_stats_and_reset())
```
Comment (ericl) [inline python/ray/rllib/optimizers/async_samples_optimizer.py]: This syntax is needed for python 2.

**Comment by ericl** [comment]
jenkins retest this please

**Comment by ericl** [comment]
Merged, thanks!

**Pull request merged**